app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -120,26 +123,33 @@ public static void showInsertTableRowDialog(final Activity activity, final boole
         SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
     }
 
-    public static void showSearchFilesDialog(Activity activity, File searchDir, Callback.a1<String> callback) {
-        SearchOrCustomTextDialog.DialogOptions dopt = new SearchOrCustomTextDialog.DialogOptions();
-        baseConf(activity, dopt);
-        dopt.callback = query -> SearchOrCustomTextDialog.recursiveFileSearch(activity, searchDir, query, (Callback.a1<List<String>>) searchResults -> {
-            dopt.callback = callback;
-            dopt.isSearchEnabled = !searchResults.isEmpty();
-            dopt.data = searchResults;
-            dopt.cancelButtonText = R.string.close;
-            dopt.titleText = R.string.select;
-            dopt.messageText = null;
-            if (dopt.data.isEmpty()) {
-                dopt.messageText = "     ¯\\_(ツ)_/¯     ";
-            }
-            SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
-        });
-        dopt.titleText = R.string.search;
-        dopt.isSearchEnabled = true;
-        dopt.messageText = activity.getString(R.string.recursive_search_in_current_directory);
-        dopt.searchHintText = R.string.search;
-        SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
+    public static void showSearchFilesDialog(Activity activity, File searchDir, Callback.a2<String, Integer> callback) {
+        if (SearchEngine.isSearchExecuting) {
+            return;
+        }
+
+        AppSettings appSettings = new AppSettings(activity);
+
+        Callback.a1<FileSearchDialog.CallBackOptions> fileSearchDialogCallback = (cb_options) -> {
+
+            final boolean isShowResultOnCancel = appSettings.isShowSearchResultOnCancel();
+            final Integer maxSearchDepth = appSettings.getSearchMaxDepth();
+            final List<String> ignoredDirs = appSettings.getIgnoredSearchDirNames();
+            final List<String> ignoredFiles = appSettings.getIgnoredSearchFileNames();
+            final boolean isShowMatchPreview = appSettings.isShowMatchPreview();
+            final List<String> searchByContentExtensions = appSettings.getSearchByContentExtensions();
+            appSettings.setSearchQueryRegexUsing(cb_options._isRegexQuery);
+            appSettings.setSearchQueryCaseSensitivity(cb_options._isCaseSensitiveQuery);
+            appSettings.setSearchInContent(cb_options._isSearchInContent);
+            appSettings.setOnlyFirstContentMatch(cb_options._isOnlyFirstContentMatch);
+            SearchEngine.Config config = new SearchEngine.Config(searchDir, cb_options._query, isShowResultOnCancel, maxSearchDepth, ignoredDirs, ignoredFiles, cb_options._isRegexQuery, cb_options._isCaseSensitiveQuery, cb_options._isSearchInContent, cb_options._isOnlyFirstContentMatch, isShowMatchPreview, searchByContentExtensions);
```

Comment:
Would it be possible to use a dopt like mechanism here? These long function calls are hard to follow

Reply:
> 
> Looks like I do not have push rights. Code is ready though. @adelobosko could you give me permission?

Sure, I sent you the invite